tc/core/polyhedral/cuda/codegen.cc

```diff
@@ -365,8 +365,32 @@ void emitReductionInit(
 }
 
 namespace {
+// RAII-style wrapper around CodegenStatementContext that wraps the output
+// streamed to context.ss into an "__ldg()" intrinsic, if the tensor
+// with the given identifier is known to be read-only.
+struct LdgWrapper {
+ public:
+  LdgWrapper(const CodegenStatementContext& context, isl::id id)
+      : readOnly_(context.readOnlySet.count(id) > 0), out_(context.ss) {
```

Comment:
Hmm... this looks like an overly complicated way of wrapping the access as
it requires the user to print the access inside a scope.
IMO, it would be more transparent if the user would print to a string first (in the case
where it's not a single string already) and then have that wrapped.
But anyway, at least the duplication is gone.

Reply:
That would require creating a new stringstream. printing to it, then taking a string from that stream and passing it to a function that prints into another stringstream.  The latter stream still being accessible in the context where the former must be used, that looked confusing.